php-checks/src/main/java/org/sonar/php/checks/wordpress/WordPressLateConfigCheck.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks.wordpress;
+
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.plugins.php.api.tree.CompilationUnitTree;
+import org.sonar.plugins.php.api.tree.declaration.CallArgumentTree;
+import org.sonar.plugins.php.api.tree.expression.BinaryExpressionTree;
+import org.sonar.plugins.php.api.tree.expression.FunctionCallTree;
+
+@Rule(key = "S6347")
+public class WordPressLateConfigCheck extends WordPressConfigVisitor {
+
+  private static final String MESSAGE = "Configuration options at this location will not be taken into account.";
+  private boolean endOfConfig;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    if (isWpConfigFile()) {
+      endOfConfig = false;
+      super.visitCompilationUnit(tree);
+    }
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    checkEndOfConfig(tree);
+    super.visitFunctionCall(tree);
+  }
+
+  @Override
+  void visitConfigDeclaration(FunctionCallTree config) {
+    if (endOfConfig) {
+      newIssue(config, MESSAGE);
+    }
+  }
+
+  private void checkEndOfConfig(FunctionCallTree tree) {
+    if ("require_once".equalsIgnoreCase(CheckUtils.functionName(tree)) && isSettingsInclusion(tree)) {
+      endOfConfig = true;
+    }
+  }
+
+  private static boolean isSettingsInclusion(FunctionCallTree tree) {
+    return CheckUtils.argument(tree, "", 0)
+      .map(CallArgumentTree::value)
+      .filter(BinaryExpressionTree.class::isInstance)
```

Comment:
This does not seem to work with `require_once(ABSPATH . 'wp-settings.php');` (with parenthesis). The reason for this seems to be that in that case the argument value will be a `ParenthesizedExpressionTree`. This is special to `require_once`, and similar "functions" in our parser (but I wouldn't change the parser now).

I think we should handle that case correctly too, as it often appears (https://sourcegraph.com/search?q=context:global+lang:PHP+wp-settings.ph&patternType=literal). In fact, WordPress only removed the parenthesis last year https://github.com/WordPress/WordPress/commit/47ed56f38ff85423c82642fa3a19b0948bf5b6a4